python/pynessie/cli.py

```diff
@@ -232,32 +234,34 @@ def log(  # noqa: C901
 ) -> None:
     """Show commit log.
 
-    REVISION_RANGE optional branch, tag or hash to start viewing log from. If of the form <hash>..<hash> only show log
-    for given range
+    REVISION_RANGE optional hash to start viewing log from. If of the form <hash>..<hash> only show log
+    for given range on the particular ref that was provided
 
     PATHS optional list of paths. If given, only show commits which affected the given paths
     """
-    if not revision_range:
-        start = ctx.nessie.get_default_branch()
-        end = None
-    else:
+    if not ref:
+        ref = ctx.nessie.get_default_branch()
+    start = None
+    end = None
+    if revision_range:
         if ".." in revision_range:
             start, end = revision_range.split("..")
         else:
             start = revision_range
-            end = None
 
     filtering_args: Any = {}
     if number:
         filtering_args["max"] = str(number)
+    if start:
```

Comment:
I think, we should have a follow-up later to push all these down to the server. Especially "start" + "end" + "max" are relatively easy to implement and "max" is relatively easy to push down via the paging functionality.

Reply:
agreed